src/main/java/org/sagebionetworks/bridge/exporter/handler/SynapseExportHandler.java

```diff
@@ -162,9 +165,21 @@ private synchronized TsvInfo initTsvForTask(ExportTask task) {
         columnDefList.addAll(getSynapseTableColumnList(task));
 
         // Create or update table if necessary.
+        boolean isExisted = true;
+
         String synapseTableId = getManager().getSynapseTableIdFromDdb(task, getDdbTableName(), getDdbTableKeyName(),
                 getDdbTableKeyValue());
-        if (synapseTableId == null) {
+
+        // check if the table in synapse currently
+        ExportWorkerManager manager = getManager();
+        SynapseHelper synapseHelper = manager.getSynapseHelper();
+        try {
+            synapseHelper.getTableWithRetry(synapseTableId);
+        } catch (SynapseNotFoundException e) {
+            isExisted = false;
+        }
```

Comment:
Minor optimization: if synapseTableId is null, synapseHelper.getTableWithRetry() is guaranteed to throw. Might as well save a server call.